File: src/e_245/DateValidator.java
package e_245;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

/**
 * Created by osk on 2016-02-21.
 */
public class DateValidator {
    public static void main(String[] args) {
        String[] input = { "2/13/2015", "1-31-10", "5 10 2015", "2012 3 17", "2001-01-01", "2008/01/07" };

        for (String s : input) {
            System.out.println(s + "\n" + reformatDate(s) + "\n");
        }
    }

    private static String reformatDate(String s) {
        Scanner scan = new Scanner(s);
        scan.useDelimiter("[^0-9]");

        int month = scan.nextInt();
        int day = scan.nextInt();
        int year = scan.nextInt();

        if (month > 12) {
            int temp = month;
            month = day;
            day = year;
            year = temp;
        }

        if (year < 100) {
            year += 2000;
        }

        LocalDate date = LocalDate.of(year, month, day);
        DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE;

        scan.close();
        return date.format(formatter);
    }
}

File: src/e_245/datevalidator.cc
#include <iostream>
#include <fstream>

using namespace std;

int main() {
	fstream inputfile("input.txt", ios_base::in);

	int a, b, c, temp;
	while (inputfile >> a) {
		inputfile.get();
		inputfile >> b;
		inputfile.get();
		inputfile >> c;

		if (a <= 12) {
			temp = a;
			a = c;
			c = b;
			b = temp;
		}
		cout << (a < 100 ? "20" : "") << a << '-' << (b < 10 ? "0" : "") << b << '-'
			<< (c < 10 ? "0" : "") << c << endl;
	}
	inputfile.close();
}